ITEST/itest.kafka.smoke/test/org/openl/itest/RunKafkaSmokeITest.java

```diff
@@ -46,200 +64,225 @@ public void testRest() {
+    public void methodSimpleFail() {
+        try (KafkaProducer<String, String> producer = createKafkaProducer(kafkaContainer.getBootstrapServers());
+             KafkaConsumer<String, String> consumer = createKafkaConsumer(kafkaContainer.getBootstrapServers())) {
+            consumer.subscribe(Collections.singletonList("hello-dlt-topic"));
+
+            producer.send(new ProducerRecord<>("hello-in-topic", "key1", "5"));
+
+            checkKafkaResponse(consumer, "key1", "5");
+
+            producer.send(new ProducerRecord<>("hello-in-topic", "key1", "{\"hour\": 22}"));
+            Unreliables.retryUntilTrue(
+                    20,
+                    TimeUnit.SECONDS,
+                    () -> {
+                        ConsumerRecords<String, String> records = consumer.poll(Duration.ofMillis(1000));
+                        if (records.isEmpty()) {
+                            return false;
+                        }
+                        assertEquals(1, records.count());
+                        ConsumerRecord<String, String> response = records.iterator().next();
+                        assertEquals(response.value(), "{\"hour\": 22}");
+                        assertEquals(response.key(), "key1");
+                        Assert.assertEquals("fail", getHeaderValue(response, KafkaHeaders.DLT_EXCEPTION_MESSAGE));
+                        return true;
+                    }
+            );
+
+            consumer.unsubscribe();
+        }
     }
 
     @Test
-    public void methodSimpleOkWithReplyTopic() throws Exception {
-        KeyValue<String, String> record = new KeyValue<>("key1", "{\"hour\": 5}");
-        record.addHeader(KafkaHeaders.REPLY_TOPIC, "hello-reply-topic", StandardCharsets.UTF_8);
-        cluster.send(SendKeyValues.to("hello-in-topic", Collections.singletonList(record)).useDefaults());
+    public void serviceSimpleOk() {
+        ProducerRecord<String, String> producerRecord = new ProducerRecord<>("hello-in-topic-2", "key1", "{\"hour\": 5}");
+        producerRecord.headers().add(KafkaHeaders.METHOD_NAME, "Hello".getBytes(StandardCharsets.UTF_8));
+        testKafka(producerRecord, "hello-out-topic-2", "Good Morning");
+    }
 
-        ObserveKeyValues<String, String> observeRequest = ObserveKeyValues.on("hello-reply-topic", 1)
-            .with("metadata.max.age.ms", 1000)
-            .with("group.id", "junit")
-            .build();
-        List<KeyValue<String, String>> observes = cluster.observe(observeRequest);
-        Assert.assertEquals(1, observes.size());
-        Assert.assertEquals("Good Morning", observes.get(0).getValue());
-        Assert.assertEquals("key1", observes.get(0).getKey());
+    @Test
+    public void serviceSimpleFail() {
+        ProducerRecord<String, String> producerRecord = new ProducerRecord<>("hello-in-topic-2", "key1", "5");
+        producerRecord.headers().add(KafkaHeaders.METHOD_NAME, "Hello".getBytes(StandardCharsets.UTF_8));

```

Comment:
A lot of similar strings in the test, Maybe is it better to extract addHeader() method?